The intention of this patch is to keep FreeBSD shared libs out of the first 4 GiB of memory in a non-aslr mode (rtld seems to always place them after the executable).
In aslr-mode the system does that implicitly (probably because that benefits randomization).
The downside is that we fix the entry main address in the aslr-mode as well, making it entirely predictable, which is against the spirit of aslr.

--- configure.ac.orig	2026-08-21 20:25:07 UTC
+++ configure.ac
@@ -608,7 +608,6 @@
     WINE_TRY_PE_CFLAGS([-Wlogical-op])
     WINE_TRY_PE_CFLAGS([-Wenum-enum-conversion],[:],WINE_TRY_PE_CFLAGS([-Wenum-conversion]))
     WINE_TRY_PE_CFLAGS([-ffunction-sections])
-    WINE_TRY_PE_CFLAGS([-ffp-exception-behavior=maytrap])
 
     dnl clang had broken -fms-hotpatch support before version 18 (https://github.com/llvm/llvm-project/pull/77245)
     WINE_TRY_PE_CFLAGS([-fms-hotpatch -DMIN_CLANG_VERSION=18],
@@ -1109,6 +1109,19 @@ case $host_os in
             ;;
         esac
         ;;
+       freebsd*)
+        case $HOST_ARCH in
+          i386|arm)
+            WINE_TRY_CFLAGS([-Wl,-Ttext-segment=0x60000000],
+                            [AS_VAR_APPEND([WINELOADER_LDFLAGS],[" -Wl,-Ttext-segment=0x60000000"])],
+                            [WINE_TRY_CFLAGS([-Wl,--section-start,.interp=0x60000400],
+                                [AS_VAR_APPEND([WINELOADER_LDFLAGS],[" -Wl,--section-start,.interp=0x60000400"])])])
+            ;;
+          *)
+            AS_VAR_APPEND([WINELOADER_LDFLAGS],[" -Wl,-Ttext-segment=0x240000000 -pie"])
+            ;;
+        esac
+        ;;
       *)
         case $HOST_ARCH in
           i386|arm)
